cogs/lastactive.py

```diff
@@ -0,0 +1,112 @@
+"""
+lastactive (Ren v2) - Tracks when a user was last active.
+
+How to use:
+    bot.last_active[server_id][channel_id][author_id] -> Retrieves datetime (UTC) of last message sent by that author in
+    that channel.
+    You can change the parameters of the object in the setup method.
+
+Last updated by jangarong on August 21st, 2019.
+"""
+import asyncio
+import pickle
+from datetime import datetime, timedelta
+
+
+class LastActive:
+
+    """
+    pickle_path : (String) Where the pickled file is saved/loaded.
+    from_pickle : (Boolean) If true, it will retrieve last active data from pickle based on path. If false, it will
+        retrieve last active data via logs from channels.
+    to_pickle : (Boolean) If true, it will pickle the last active data and save it in the desired path periodically.
+    limit : (Integer) Number of messages read in the chat history of each channel. Keep in mind that the larger the
+        number, the longer it will take for the bot to set up.
+
+    To change these values, see the setup function down below.
+    """
+    def __init__(self, bot, pickle_path='last_active.pkl', from_pickle=False, to_pickle=True, limit=500):
+        self.bot = bot
+        self.bot.last_active = {}
+        self.pickle_path = pickle_path
+        self.from_pickle = from_pickle
+        self.to_pickle = to_pickle
+        self.limit = limit
+
+        # create loop that goes per day
+        self.lastChecked = datetime.now() - timedelta(days=1)
+        self.bgTask = self.bot.loop.create_task(self.pickle_loop())
+
+    def _to_pickle(self):
```

Comment:
I didn't realize you were doing this yourself, perhaps you can look at using `config` within the `cogs/utils` directory, or maybe even dataIO (see other cogs for this as well).  Thoughts @KaguneAstra?